simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,395 @@
+"""
+Generates data for the simple-acquaintances PSL example.
+
+Use -h/--help for usage options.
+"""
+
+import argparse
+import json
+import numpy as np
+import os
+
+from itertools import combinations
+from scipy.stats import truncnorm
+from sklearn.metrics.pairwise import cosine_similarity
+
+DEFAULT_GLOBAL_LIKES = 5
+DEFAULT_LOCAL_LIKES = 5
+DEFAULT_LOCAL_LIKES_VARIANCE = 1.0
+DEFAULT_PEOPLE = 25
+DEFAULT_PLACES = 5
+DEFAULT_PLACES_LIVED_MEAN = 3.0
+DEFAULT_PLACES_LIVED_SD = 1.0
+DEFAULT_TARGET_SPLIT = 0.2
+
+
+def checkNonNegativeInt(arg):
+    """
+    Check if argument is non negative integer.
+    """
+    if not arg.isdigit():
+        raise argparse.ArgumentTypeError(
+            '{} is not a non negative integer.'.format(arg))
+    return int(arg)
+
+
+def checkValidSplit(arg):
+    """
+    Check if argument is a valid split for targets i.e {0...1}.
+    """
+    try:
+        if 0 <= float(arg) <= 1:
+            return arg
+        else:
+            raise argparse.ArgumentTypeError('The input should be a valid float between [0, 1].')
+    except ValueError as e:
+        raise argparse.ArgumentTypeError('{}. The input should be a valid float between [0, 1].'
+                                        .format(e))
+
+
+def checkNonNegativeFloat(arg):
+    """
+    Check if argument is non negative float.
+    """
+    try:
+        if 0 <= float(arg):
+            return arg
+        else:
+            raise argparse.ArgumentTypeError('{} is not a non negative float.'.format(arg))
+    except ValueError as e:
+        raise argparse.ArgumentTypeError('{}. The input should be a non negative float value.'
+                                        .format(e))
+
+
+def parse_args():
+    """
+    Parse CLI arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description='Generate data for the PSL simple-acquaintances example.')
+
+    parser.add_argument(
+        '--coinFlipSeed',
+        default = None,
+        type = float,
```

Comment:
Seeds are always non-floating point types so they are reproducible.
If seeds were floats then languages with different internal representations of floats (like 32 vs 64 bits, or different implementations of IEEE-754) would get different random numbers with the same RNG algorithm.
Python uses ints for seeds (long is used for languages that have them (except Python 2)).